OpenRA.Game/Graphics/RgbaColorRenderer.cs

```diff
@@ -0,0 +1,129 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.Drawing;
+
+namespace OpenRA.Graphics
+{
+	public class RgbaColorRenderer : Renderer.IBatchRenderer
+	{
+		static readonly float2 Offset = new float2(0.5f, 0.5f);
+
+		readonly Renderer renderer;
+		readonly IShader shader;
+		readonly Action renderAction;
+
+		readonly Vertex[] vertices;
+		int nv = 0;
+
+		public RgbaColorRenderer(Renderer renderer, IShader shader)
+		{
+			this.renderer = renderer;
+			this.shader = shader;
+			vertices = new Vertex[renderer.TempBufferSize];
+			renderAction = () => renderer.DrawBatch(vertices, nv, PrimitiveType.QuadList);
+		}
+
+		public void Flush()
+		{
+			if (nv > 0)
+			{
+				renderer.Device.SetBlendMode(BlendMode.Alpha);
+				shader.Render(renderAction);
+				renderer.Device.SetBlendMode(BlendMode.None);
+
+				nv = 0;
+			}
+		}
+
+		public void DrawLine(float2 start, float2 end, float width, Color color)
+		{
+			renderer.CurrentBatchRenderer = this;
+
+			if (nv + 4 > renderer.TempBufferSize)
+				Flush();
+
+			var delta = (end - start) / (end - start).Length;
+			var corner = width / 2 * new float2(-delta.Y, delta.X);
+
+			color = Util.PremultiplyAlpha(color);
+			var r = color.R / 255.0f;
```

Comment:
This math could have been done manually and pushed back into the (new) Color, yes?